parlai/core/image_featurizers.py

```diff
@@ -57,13 +57,35 @@ def init_cnn(self, opt):
             print('[ Using CUDA ]')
             torch.cuda.set_device(opt.get('gpu', -1))
 
-        cnn_type, layer_num = self._image_mode_switcher()
-
-        # initialize the pretrained CNN using pytorch.
-        CNN = getattr(torchvision.models, cnn_type)
-
-        # cut off the additional layer.
-        self.netCNN = nn.Sequential(*list(CNN(pretrained=True).children())[:layer_num])
+        if 'resnet' in self.image_mode:
+            cnn_type, layer_num = self._image_mode_switcher()
+            # initialize the pretrained CNN using pytorch.
+            CNN = getattr(torchvision.models, cnn_type)
+
+            # cut off the additional layer.
+            self.netCNN = nn.Sequential(
+                *list(CNN(pretrained=True).children())[:layer_num])
+        elif 'resnext' in self.image_mode:
+            try:
+                model = torch.hub.load('facebookresearch/WSL-Images', self.image_mode)
+                self.netCNN = nn.Sequential(
+                    *list(model.children())[:-1])
+            except RuntimeError as e:
+                # Perhaps specified one of the wrong model names
+                print('If you have specified one of the resnext101 wsl models, '
+                      'please make sure it is one of the following: \n'
+                      'resnext101_32x8d_wsl, resnext101_32x16d_wsl, '
+                      'resnext101_32x32d_wsl, resnext101_32x48d_wsl')
+                raise e
+            except AttributeError:
+                # E.g. "module 'torch' has no attribute 'hub'"
+                raise RuntimeError(
+                    'Please install the latest pytorch distribution to have access '
```

Comment:
Do you know what version needs it? It'd be nice to say it explicitly.

Reply:
yes